Source/Workers/decodeDracoPointCloud.js

```diff
@@ -0,0 +1,212 @@
+            vertexArray = new Uint8Array(vertexArrayLength);
+            dracoDecoder.GetAttributeUInt8ForAllPoints(dracoGeometry, attribute, attributeData);
+        } else {
+            attributeData = new draco.DracoUInt16Array();
+            vertexArray = new Uint16Array(vertexArrayLength);
+            dracoDecoder.GetAttributeUInt16ForAllPoints(dracoGeometry, attribute, attributeData);
+        }
+
+        for (var i = 0; i < vertexArrayLength; ++i) {
+            vertexArray[i] = attributeData.GetValue(i);
+        }
+
+        draco.destroy(attributeData);
+        return vertexArray;
+    }
+
+    function decodeDracoTypedArray(dracoGeometry, attribute, vertexArrayLength) {
+        var vertexArray;
+        var attributeData;
+
+        // Some attribute types are casted down to 32 bit since Draco only returns 32 bit values
+        switch (attribute.data_type()) {
+            case 1: case 11: // DT_INT8 or DT_BOOL
+                attributeData = new draco.DracoInt8Array();
+                vertexArray = new Int8Array(vertexArrayLength);
+                dracoDecoder.GetAttributeInt8ForAllPoints(dracoGeometry, attribute, attributeData);
+                break;
+            case 2: // DT_UINT8
+                attributeData = new draco.DracoUInt8Array();
+                vertexArray = new Uint8Array(vertexArrayLength);
+                dracoDecoder.GetAttributeUInt8ForAllPoints(dracoGeometry, attribute, attributeData);
+                break;
+            case 3: // DT_INT16
+                attributeData = new draco.DracoInt16Array();
+                vertexArray = new Int16Array(vertexArrayLength);
+                dracoDecoder.GetAttributeInt16ForAllPoints(dracoGeometry, attribute, attributeData);
+                break;
+            case 4: // DT_UINT16
+                attributeData = new draco.DracoUInt16Array();
+                vertexArray = new Uint16Array(vertexArrayLength);
+                dracoDecoder.GetAttributeUInt16ForAllPoints(dracoGeometry, attribute, attributeData);
+                break;
+            case 5: case 7: // DT_INT32 or DT_INT64
+                attributeData = new draco.DracoInt32Array();
+                vertexArray = new Int32Array(vertexArrayLength);
+                dracoDecoder.GetAttributeInt32ForAllPoints(dracoGeometry, attribute, attributeData);
+                break;
+            case 6: case 8: // DT_UINT32 or DT_UINT64
+                attributeData = new draco.DracoUInt32Array();
+                vertexArray = new Uint32Array(vertexArrayLength);
+                dracoDecoder.GetAttributeUInt32ForAllPoints(dracoGeometry, attribute, attributeData);
+                break;
+            case 9: case 10: // DT_FLOAT32 or DT_FLOAT64
+                attributeData = new draco.DracoFloat32Array();
+                vertexArray = new Float32Array(vertexArrayLength);
+                dracoDecoder.GetAttributeFloatForAllPoints(dracoGeometry, attribute, attributeData);
+                break;
```

Comment:
This seems like a lot of duplicate code with `decodeDraco`, and this function in particular is identical.  Can we share this somehow?